src/metabase/pulse/render/body.clj

```diff
@@ -601,66 +567,98 @@
+
+(defn- render-multiple-lab-chart
+  "When multiple non-scalar cards are combined, render them as a line, area, or bar chart"
+  [render-type card dashcard {:keys [viz-settings]
+                              :as   data}]
+  (let [viz-settings      (merge viz-settings (:visualization_settings dashcard))
+        multi-res         (pu/execute-multi-card card dashcard)
+        ;; multi-res gets the other results from the set of multis.
+        ;; we shove cards and data here all together below for uniformity's sake
+        viz-settings      (set-default-stacked viz-settings card)
+        viz-settings-seqs (cons viz-settings (map (comp :visualization_settings :card) multi-res))
+        cards             (cons card (map :card multi-res))
+        multi-data        (cons data (map #(get-in % [:result :data]) multi-res))
+        axis-row-fns      (fn [card data]
+                            [(or (ui-logic/mult-x-axis-rowfn card data) #(vector (first %)))
+                             (or (ui-logic/mult-y-axis-rowfn card data) #(vector (second %)))])
+        rowfns            (mapv axis-row-fns cards multi-data)
+        row-seqs          (for [[row-seq [x-rowfn y-rowfn]] (map vector (map :rows multi-data) rowfns)]
+                            (map (comp flatten (juxt x-rowfn y-rowfn))
+                                 (common/row-preprocess x-rowfn y-rowfn row-seq)))
+        col-seqs          (map :cols multi-data)
+        first-rowfns      (first rowfns)
+        [[x-col] [y-col]] ((juxt (first first-rowfns) (second first-rowfns)) (first col-seqs))
+        labels            (x-and-y-axis-label-info x-col y-col viz-settings)
+        names             (map :name cards)
+        types             (replace {:scalar :bar} (map :display cards))
+        settings-seqs     (map (partial ->ts-viz x-col y-col labels) viz-settings-seqs)
+        y-axis-positions  (take (count names) (default-y-pos viz-settings))
+        series            (for [[enforced-type rows cols viz-settings [x-axis-rowfn y-axis-rowfn] y-axis-position card-name]
+                                (map vector types row-seqs col-seqs viz-settings-seqs rowfns y-axis-positions names)]
+                            (let [
+                                  x-cols      (x-axis-rowfn cols)
+                                  y-cols      (y-axis-rowfn cols)
+                                  x-rows      (map x-axis-rowfn rows)
+                                  y-rows      (map y-axis-rowfn rows)
+                                  joined-rows (map vector x-rows y-rows)]
+                              (map
+                               #(assoc % :yAxisPosition y-axis-position)
+                               (if (= (count x-cols) 1)
+                                 (single-x-axis-combo-series enforced-type joined-rows x-cols y-cols viz-settings card-name)
+                                 (double-x-axis-combo-series enforced-type joined-rows x-cols y-cols viz-settings card-name)))))]
```

Comment:
This is another major change that fix multiple series dashcard not rendering all series. We need to reuse `single-x-axis-combo-series` and `double-x-axis-combo-series` to make sure we get all the series inside each question when a question has more than 1 series i.e. when a question has multiple metrics or has 2 dimensions (which results in multiple breakout values)